src/navigator/DrawerNavigator.tsx

```diff
@@ -264,7 +272,9 @@ export default function DrawerNavigator() {
       {!!dappsListUrl && (
         <Drawer.Screen
           name={Screens.DAppsExplorerScreen}
-          component={DAppsExplorerScreenLegacy}
+          component={
```

Comment:
i noticed with this approach, if the app is already loaded and the remote config values change, this change isn't applied until the app is completely reloaded. e.g. if you load the app without filters, then the remote config value changes (in the slice.ts, change the value of dappsFilterEnabled and save), the screen component is already registered in the drawer so it doesn't change?

i think instead of doing this here, the component can always be `DAppsExplorerScreen` and within `DAppsExplorerScreen`, we can do the split. we did something similar with CPV https://github.com/valora-inc/wallet/blob/06e188c9d69325d25eb2ff84bd413f3a83364bf8/src/verify/VerificationEducationScreen.tsx#L68-L76 - the benefit is the user can use the filters on the first session that the remote config value updates (instead of the second) and when we remove the legacy code, we don't have to touch the navigator

Reply:
Made this change in ac6f9bd7f74e9a0e9f7bfbe92f5c4dbfb80dba19! This will make implementing the search a lot easier. There might be a bit of code duplication, but I agree it's a better experience for users.

Reply:
After this change when developing locally I am seeing a possible bug. 
Reproduction Steps.

1. Set `dappsFilterEnabled = true` in `remoteConfigValuesDefaults.ts`.
2. Launch the app and navigate to the dapps explorer.
3. Observe `DappsExplorerScreenFilter` is served and WAI.
4. Set the `dappsFilterEnabled = false` in `remoteConfigValuesDefaults.ts`.
5. Navigate to the dapps explorer.
6. Observe `DappsExplorerScreenFilter` is served incorrectly and the dapps are not displayed. This probably because we are fetching the V1 dapp list but attempting to display the results in the DappExplorerScreenFilter.tsx which expects v2.

Reply:
Removed this change in 48909fcb59866c42bfc08c3c71b11a34ae3c42a4.